### PR TITLE
[MI-2128]: Added logic to auto select dropdown values based on best scenario and improvements.

### PR DESCRIPTION
**Pull request by avas27JTG**
Added logic to auto select dropdown values based on best scenario and improvements.

### CODE REVIEW

**Review by ayusht2810 — APPROVED** [review]

File: webapp/src/containers/modals/SubscribeModal/index.tsx [hunk 1]
```diff
@@ -163,23 +163,23 @@ const SubscribeModal = () => {
     useEffect(() => {
         if (isChannelListSuccess && !showResultPanel) {
             setChannelOptions(channelList?.map((channel) => ({
-                label: <span><i className={`icon ${channel.type === 'P' ? 'icon-lock-outline' : 'icon-globe'} dropdown-option-icon`}/>{channel.display_name}</span>,
+                label: <span><i className={`icon ${channel.type === plugin_constants.common.channelType.priivate ? 'icon-lock-outline' : 'icon-globe'} dropdown-option-icon`}/>{channel.display_name}</span>,
```
Comment (sooraj-shukla) [inline webapp/src/containers/modals/SubscribeModal/index.tsx:166]:
```suggestion
                label: <span><i className={`icon ${channel.type === plugin_constants.common.channelType.private ? 'icon-lock-outline' : 'icon-globe'} dropdown-option-icon`}/>{channel.display_name}</span>,
```

**Review by sooraj-shukla — APPROVED** [review]

File: webapp/src/containers/modals/SubscribeModal/index.tsx [hunk 2]
```diff
@@ -41,8 +40,9 @@ const SubscribeModal = () => {
         makeApiRequestWithCompletionStatus,
         state,
     } = usePluginApi();
-    const {visibility} = getSubscribeModalState(state);
-    const {currentTeamId} = useSelector((reduxState: GlobalState) => reduxState.entities.teams);
+    const {visibility, project, organization} = getSubscribeModalState(state);
+    const {currentTeamId} = useSelector((pluginState: GlobalState) => pluginState.entities.teams);
```
Comment (manojmalik20) [inline webapp/src/containers/modals/SubscribeModal/index.tsx:44]:
```suggestion
    const {currentTeamId} = useSelector((state: GlobalState) => pluginState.entities.teams);
```
I don't think the `pluginState` is the right variable name here.

File: webapp/src/containers/modals/TaskModal/index.tsx [hunk 3]
```diff
@@ -178,9 +178,6 @@ const TaskModal = () => {
             error={showApiErrorMessages(isError, error as ApiErrorResponse)}
         >
             <>
-                {
```
Comment (manojmalik20) [inline webapp/src/containers/modals/TaskModal/index.tsx:181]: Why have you removed the loader from everywhere?

File: webapp/src/types/common/store.d.ts [hunk 4]
```diff
@@ -15,6 +15,7 @@ type GlobalModalState = {
 type GlobalModalActionPayload = {
     isVisible: boolean
     commandArgs: Array<string>
+    args?: Array<string>
```
Comment (manojmalik20) [inline webapp/src/types/common/store.d.ts:18]: What's the main difference between commandArgs and args here?

Comment (avas27JTG) [reply webapp/src/types/common/store.d.ts:18]: `commandArgs` will come from slash command and args will come from anywhere like dispatching action on button click, etc

File: webapp/src/containers/modals/SubscribeModal/index.tsx (hunk 2 above)
Comment (manojmalik20) [reply webapp/src/containers/modals/SubscribeModal/index.tsx:45]: And what about the previous variable `reduxState`?

**Review by manojmalik20 — APPROVED** [review]

File: webapp/src/containers/modals/SubscribeModal/index.tsx (hunk 2 above)
Comment (avas27JTG) [reply webapp/src/containers/modals/SubscribeModal/index.tsx:45]: yes can use that

**Pull request merged**